### PR TITLE
Feature/JC19-456-initialize-anime-app-and-authorization

### PR DESCRIPTION
**Pull request by BKrasnov**

### CODE REVIEW

File: apps/react/src/api/services/tokenService.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import { LocalStorageService } from './localStorageService';
+
+const TOKEN = 'token';
+
+/** Functionality for working with Token. */
+export namespace TokenService {
+
+  /** Checks if the token exists in local storage. */
+  export function isToken(): boolean {
```
Comment (timofei-maximov-saritasa) [inline apps/react/src/api/services/tokenService.ts:11]: Function name sounds to me like it is a guard function. Let's improve naming

File: apps/react/src/features/auth/components/LoginForm/LoginForm.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+              isLoading ? (
+                <CircularProgress />
+              ) : (
+                <Button type="submit" variant="contained" sx={{ mt: 3, mb: 2 }}>
+                  Sign In
+                </Button>
+              )
+            }
+            <Typography component='span'>
+              <Link to={'../register'}>
```
Comment (timofei-maximov-saritasa) [inline apps/react/src/features/auth/components/LoginForm/LoginForm.tsx:81]: Actually, you can use absolute path here `/register`
```suggestion
              <Link to="../register">
```

File: apps/react/src/features/auth/components/LoginForm/LoginForm.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                {'No have an account? Sign Up'}
```
Comment (timofei-maximov-saritasa) [inline apps/react/src/features/auth/components/LoginForm/LoginForm.tsx:82]:
```suggestion
                No have an account? Sign Up
```
Please check your grammar, there is a mistake in your sentence.

File: apps/react/src/features/auth/components/LoginForm/form-settings.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { Login } from '@js-camp/core/models/login';
```
Comment (timofei-maximov-saritasa) [inline apps/react/src/features/auth/components/LoginForm/form-settings.ts:1]: Please, be consistent with file names. You use `kebab-case` for this file, but other files use `PascalCase` or `camelCase`

File: apps/react/src/features/auth/components/RegisterForm/form-settings.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import { Registration } from '@js-camp//core/models/registration';
+import * as Yup from 'yup';
+
+/** Login form. */
+export type RegisterFormValue = Registration;
```
Comment (timofei-maximov-saritasa) [inline apps/react/src/features/auth/components/RegisterForm/form-settings.ts:5]: Can you explain a why do you need this type?

File: apps/react/src/features/auth/pages/LoginPage/LoginPage.tsx [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import { memo, FC } from 'react';
+
+import { LoginForm } from '../../components/LoginForm/LoginForm';
+
+const LoginPageComponent: FC = () => (
+  <>
+    <LoginForm />
+  </>
```
Comment (timofei-maximov-saritasa) [inline apps/react/src/features/auth/pages/LoginPage/LoginPage.tsx:8]: No need for `Fragment`(`<>`) here.
```suggestion
  <LoginForm />
```

File: apps/react/src/routes/guards/authGuard.tsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+import { useAppSelector } from '../../store';
+
+const AuthGuardComponent = () => {
+  const user = useAppSelector(selectUser);
+  const redirect: To = {
+    pathname: '/login',
+  };
+
+  if (!user) {
```
Comment (timofei-maximov-saritasa) [inline apps/react/src/routes/guards/authGuard.tsx:13]: https://wiki.saritasa.rocks/frontend/languages/ts-js/best-practices#dont-use-implicit-type-casting

File: apps/react/src/store/auth/dispatchers.ts [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
+import { AuthService } from '../../api/services/authService';
+import { UserService } from '../../api/services/userService';
+
+export const fetchUser = createAsyncThunk('auth/fetchUser', () => UserService.getUser());
+
+export const authLogin = createAsyncThunk(
+  'auth/login',
+  async(loginData: Login, thunkApi) => {
```
Comment (timofei-maximov-saritasa) [inline apps/react/src/store/auth/dispatchers.ts:12]: Missing type annotation for `thunkApi`. Also, you can use object destructuring

File: apps/react/src/store/auth/selectors.ts [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+
+import { RootState } from '../store';
+
+export const selectUser = createSelector((state: RootState) => state.auth.user, user => user);
+
+export const selectUserLoading = createSelector((state: RootState) => state.auth.isLoading, isLoading => isLoading);
+
+export const selectUserError = createSelector((state: RootState) => state.auth.error, error => error);
+
+export const selectUserLoggedIn = createSelector((state: RootState) => state.auth.isLoggedIn, isLoggedIn => isLoggedIn);
```
Comment (timofei-maximov-saritasa) [inline apps/react/src/store/auth/selectors.ts:11]: Don't forget to add docs for your selectors.

File: apps/react/src/routes/guards/nonAuthGuard.tsx [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+import { useAppSelector } from '../../store';
+
+const NonAuthGuardComponent: FC = () => {
+  const user = useAppSelector(selectUser);
+
+  const redirect: To = {
+    pathname: '/',
+  };
+  if (user != null) {
```
Comment (timofei-maximov-saritasa) [inline apps/react/src/routes/guards/nonAuthGuard.tsx:14]: `user` has type `User | null`, so use strict inequality

**Comment by artem-sotnichenko-saritasa** [comment]
Deployed preview doesn't work, because "api-key" header is empty. I'm not sure, but the only way to fix it, is to store .env in repo

File: apps/react/src/features/auth/pages/RegisterPage/RegisterPage.tsx [hunk 11]
```diff
@@ -0,0 +1,11 @@
+import { memo, FC } from 'react';
+
+import { RegisterForm } from '../../components/RegisterForm';
+
+const RegisterPageComponent: FC = () => (
+  <>
+    <RegisterForm />
+  </>
```
Comment (artem-sotnichenko-saritasa) [inline apps/react/src/features/auth/pages/RegisterPage/RegisterPage.tsx:8]:
```suggestion
  <RegisterForm />
```

File: apps/react/src/store/auth/slice.ts [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+  name: 'auth',
+  initialState,
+  reducers: {},
+  extraReducers: builder => builder
+    .addCase(authLogin.pending, state => {
+      state.isLoading = true;
+    })
+    .addCase(authLogin.fulfilled, (state, action) => {
+      state.isLoading = false;
+      state.isLoggedIn = true;
```
Comment (artem-sotnichenko-saritasa) [inline apps/react/src/store/auth/slice.ts:16]: Don't forget setting it to false, when user is null

File: apps/react/src/features/auth/components/LoginForm/LoginForm.tsx [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+
+import { TextField } from 'formik-mui';
+
+import { useAppDispatch, useAppSelector } from '@js-camp/react/store/store';
+
+import { authLogin } from '@js-camp/react/store/auth/dispatchers';
+
+import { selectUserError, selectUserLoading } from '@js-camp/react/store/auth/selectors';
+
+import { initValues, loginFormSchema, LoginFormValue } from './form-settings';
```
Comment (artem-sotnichenko-saritasa) [inline apps/react/src/features/auth/components/LoginForm/LoginForm.tsx:16]: Please pack imports tighter, in this and other files too

**Review by artem-sotnichenko-saritasa — APPROVED** [review]

File: apps/react/src/features/auth/components/RegisterForm/form-settings.ts (hunk 5 above)
Comment (BKrasnov) [reply apps/react/src/features/auth/components/RegisterForm/form-settings.ts:5]: In RegisterForm.tsx I type values. 31 line.

**Pull request closed without merging**